File: mobile_apps/ALADDIN_Android/app/src/main/java/family/aladdin/android/ui/screens/GamesParentalControlScreen.kt
package family.aladdin.android.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.unit.dp
import family.aladdin.android.ui.components.navigation.ALADDINTopBar
import family.aladdin.android.ui.theme.*

/// 🎮 Games Parental Control Screen
/// Панель управления геймификацией
/// Источник дизайна: /mobile/wireframes/14c_games_parental_control.html
@Composable
fun GamesParentalControlScreen(
    navController: androidx.navigation.NavHostController,
    onBackClick: () -> Unit = {}
) {
    var isWheelEnabled by remember { mutableStateOf(true) }
    var isTournamentEnabled by remember { mutableStateOf(true) }
    var isUniverseEnabled by remember { mutableStateOf(true) }
    var wheelFrequency by remember { mutableStateOf(1f) }
    var prizeSector1 by remember { mutableStateOf(5f) }
    var prizeSector2 by remember { mutableStateOf(10f) }
    var prizeSector3 by remember { mutableStateOf(20f) }
    var prizeSector4 by remember { mutableStateOf(50f) }
    var prizeSector5 by remember { mutableStateOf(100f) }
    var prizeSector6 by remember { mutableStateOf(500f) }
    
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Brush.linearGradient(listOf(GradientStart, GradientMiddle, GradientEnd)))
    ) {
        Column {
            ALADDINTopBar(
                title = "УПРАВЛЕНИЕ ИГРАМИ",
                subtitle = "Родительский контроль",
                onBackClick = { navController.popBackStack() }
            )
            
            Column(
                modifier = Modifier
                    .verticalScroll(rememberScrollState())
                    .padding(top = Spacing.M),
                verticalArrangement = Arrangement.spacedBy(Spacing.L)
            ) {
                // Информация
                InfoCard()
                
                // Колесо удачи
                WheelGameCard(
                    isEnabled = isWheelEnabled,
                    onToggle = { isWheelEnabled = it },
                    frequency = wheelFrequency,
                    onFrequencyChange = { wheelFrequency = it },
                    prizes = listOf(prizeSector1, prizeSector2, prizeSector3, prizeSector4, prizeSector5, prizeSector6),
                    onPrizeChange = { index, value ->
                        when (index) {
                            0 -> prizeSector1 = value
                            1 -> prizeSector2 = value
                            2 -> prizeSector3 = value
                            3 -> prizeSector4 = value
                            4 -> prizeSector5 = value
                            5 -> prizeSector6 = value
                        }
                    }
                )
                
                // Семейный турнир
                GameCard(
                    icon = "🏆",
                    title = "Семейный турнир",
                    description = "Еженедельное соревнование всей семьи. 5 типов турниров. Призы: 🥇 +50 • 🥈 +30 • 🥉 +20 🦄",
                    isEnabled = isTournamentEnabled,
                    onToggle = { isTournamentEnabled = it }
                )
                
                // Единорог-питомец (всегда ВКЛ)
                PetGameCard()
                
                // Единорог-вселенная
                GameCard(
                    icon = "🌳",
                    title = "Единорог-вселенная",
                    description = "Сад единорогов, коллекция 10 видов, сторителлинг 5 глав.",
                    isEnabled = isUniverseEnabled,
                    onToggle = { isUniverseEnabled = it }
                )
                
                // Быстрые действия
                QuickActionsSection(
                    onDisableAll = {
                        isWheelEnabled = false
                        isTournamentEnabled = false
                        isUniverseEnabled = false
                    },
                    onEnableAll = {
                        isWheelEnabled = true
                        isTournamentEnabled = true
                        isUniverseEnabled = true
                    }
                )
                
                // Текущие настройки
                CurrentSettingsCard(
                    isWheelEnabled, wheelFrequency.toInt(),
                    listOf(prizeSector1.toInt(), prizeSector2.toInt(), prizeSector3.toInt(), prizeSector4.toInt(), prizeSector5.toInt(), prizeSector6.toInt()),
                    isTournamentEnabled, isUniverseEnabled
                )
                
                // Кнопка сохранить
                SaveButton(onBackClick)
                
                Spacer(modifier = Modifier.height(Spacing.XXL))
            }
        }
    }
}

@Composable
private fun InfoCard() {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.ScreenPadding),
        shape = RoundedCornerShape(CornerRadius.Medium),
        color = PrimaryBlue.copy(alpha = 0.15f)
    ) {
        Row(
            modifier = Modifier.padding(Spacing.M),
            horizontalArrangement = Arrangement.spacedBy(Spacing.M)
        ) {
            Text("💡", style = MaterialTheme.typography.titleLarge)
            Column {
                Text("Родительский контроль игр", style = MaterialTheme.typography.titleMedium, color = TextPrimary)
                Text("Здесь вы можете включать/отключать игровые элементы. Единорог-питомец всегда активен! 🦄", style = MaterialTheme.typography.bodySmall, color = TextSecondary)
            }
        }
    }
}

@Composable
private fun WheelGameCard(
    isEnabled: Boolean,
    onToggle: (Boolean) -> Unit,
    frequency: Float,
    onFrequencyChange: (Float) -> Unit,
    prizes: List<Float>,
    onPrizeChange: (Int, Float) -> Unit
) {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.ScreenPadding),
        shape = RoundedCornerShape(CornerRadius.Large),
        color = BackgroundMedium.copy(alpha = if (isEnabled) 0.5f else 0.3f)
    ) {
        Column(
            modifier = Modifier.padding(Spacing.M),
            verticalArrangement = Arrangement.spacedBy(Spacing.M)
        ) {
            Row(horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                Row(horizontalArrangement = Arrangement.spacedBy(Spacing.XS)) {
                    Text("🎰", style = MaterialTheme.typography.titleLarge)
                    Text("Колесо удачи", style = MaterialTheme.typography.titleMedium, color = TextPrimary)
                }
                
                Switch(checked = isEnabled, onCheckedChange = onToggle)
            }
            
            Text("Ребёнок крутит колесо и получает случайный приз от 5 до 500 🦄.", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
            
            if (isEnabled) {
                Column(verticalArrangement = Arrangement.spacedBy(Spacing.S)) {
                    Text("Частота вращений в день:", style = MaterialTheme.typography.bodySmall, color = TextPrimary)
                    
                    Slider(
                        value = frequency,
                        onValueChange = onFrequencyChange,
                        valueRange = 1f..7f,
                        steps = 5,
                        colors = SliderDefaults.colors(thumbColor = SuccessGreen, activeTrackColor = SuccessGreen)
                    )
                    
                    Row(horizontalArrangement = Arrangement.SpaceBetween) {
                        Text("1 раз", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
                        Text("${frequency.toInt()} раз в день", style = MaterialTheme.typography.bodySmall.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold), color = SuccessGreen)
                        Text("7 раз", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
                    }
                    
                    Text("⚙️ Настройка призов на барабане:", style = MaterialTheme.typography.bodySmall, color = TextPrimary)
                    
                    PrizeSlider(1, "40%", prizes[0], 1f..50f) { onPrizeChange(0, it) }
                    PrizeSlider(2, "30%", prizes[1], 5f..100f) { onPrizeChange(1, it) }
                    PrizeSlider(3, "15%", prizes[2], 10f..150f) { onPrizeChange(2, it) }
                    PrizeSlider(4, "10%", prizes[3], 20f..200f) { onPrizeChange(3, it) }
                    PrizeSlider(5, "4%", prizes[4], 50f..300f) { onPrizeChange(4, it) }
                    PrizeSlider(6, "1% ДЖЕКПОТ", prizes[5], 100f..1000f) { onPrizeChange(5, it) }
                }
            }
        }
    }
}

@Composable
private fun PrizeSlider(sector: Int, chance: String, value: Float, range: ClosedFloatingPointRange<Float>, onValueChange: (Float) -> Unit) {
    Column {
        Row(horizontalArrangement = Arrangement.SpaceBetween, modifier = Modifier.fillMaxWidth()) {
            Text("${sector}️⃣ Сектор $sector ($chance):", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
            Text("${value.toInt()} 🦄", style = MaterialTheme.typography.bodySmall.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.Bold), color = androidx.compose.ui.graphics.Color(0xFFFFD700))
        }
        Slider(value = value, onValueChange = onValueChange, valueRange = range, colors = SliderDefaults.colors(thumbColor = androidx.compose.ui.graphics.Color(0xFFFFD700), activeTrackColor = androidx.compose.ui.graphics.Color(0xFFFFD700)))
    }
}

@Composable
private fun GameCard(icon: String, title: String, description: String, isEnabled: Boolean, onToggle: (Boolean) -> Unit) {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.ScreenPadding),
        shape = RoundedCornerShape(CornerRadius.Large),
        color = BackgroundMedium.copy(alpha = if (isEnabled) 0.5f else 0.3f)
    ) {
        Column(
            modifier = Modifier.padding(Spacing.M),
            verticalArrangement = Arrangement.spacedBy(Spacing.M)
        ) {
            Row(horizontalArrangement = Arrangement.SpaceBetween) {
                Row(horizontalArrangement = Arrangement.spacedBy(Spacing.XS)) {
                    Text(icon, style = MaterialTheme.typography.titleLarge)
                    Text(title, style = MaterialTheme.typography.titleMedium, color = TextPrimary)
                }
                Switch(checked = isEnabled, onCheckedChange = onToggle)
            }
            Text(description, style = MaterialTheme.typography.labelSmall, color = TextSecondary)
        }
    }
}

@Composable
private fun PetGameCard() {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.ScreenPadding),
        shape = RoundedCornerShape(CornerRadius.Large),
        color = UnicornPurple.copy(alpha = 0.1f),
        border = androidx.compose.foundation.BorderStroke(1.dp, UnicornPurple)
    ) {
        Column(
            modifier = Modifier.padding(Spacing.M),
            verticalArrangement = Arrangement.spacedBy(Spacing.M)
        ) {
            Row(horizontalArrangement = Arrangement.SpaceBetween) {
                Row(horizontalArrangement = Arrangement.spacedBy(Spacing.XS)) {
                    Text("🦄", style = MaterialTheme.typography.titleLarge)
                    Text("Единорог-питомец", style = MaterialTheme.typography.titleMedium, color = TextPrimary)
                }
                
                Text("🔒 ВСЕГДА ВКЛ", style = MaterialTheme.typography.labelSmall, color = UnicornPurple, modifier = Modifier.padding(horizontal = 10.dp, vertical = 5.dp).background(UnicornPurple.copy(alpha = 0.3f), RoundedCornerShape(20.dp)))
            }
            Text("Тамагочи с индикаторами ❤️🍎⭐😊. Нельзя отключить - основа мотивации!", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
        }
    }
}

@Composable
private fun QuickActionsSection(onDisableAll: () -> Unit, onEnableAll: () -> Unit) {
    Column(
        modifier = Modifier.padding(horizontal = Spacing.ScreenPadding),
        verticalArrangement = Arrangement.spacedBy(Spacing.S)
    ) {
        Surface(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(CornerRadius.Medium),
            color = DangerRed.copy(alpha = 0.2f),
            border = androidx.compose.foundation.BorderStroke(2.dp, DangerRed),
            onClick = onDisableAll
        ) {
            Text("Отключить все (кроме 🦄)", style = MaterialTheme.typography.titleMedium.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold), color = DangerRed, modifier = Modifier.padding(Spacing.M))
        }
        
        Surface(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(CornerRadius.Medium),
            color = SuccessGreen.copy(alpha = 0.2f),
            border = androidx.compose.foundation.BorderStroke(2.dp, SuccessGreen),
            onClick = onEnableAll
        ) {
            Text("Включить все", style = MaterialTheme.typography.titleMedium.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold), color = SuccessGreen, modifier = Modifier.padding(Spacing.M))
        }
    }
}

@Composable
private fun CurrentSettingsCard(
    isWheelEnabled: Boolean, wheelFrequency: Int, prizes: List<Int>,
    isTournamentEnabled: Boolean, isUniverseEnabled: Boolean
) {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.ScreenPadding),
        shape = RoundedCornerShape(CornerRadius.Medium),
        color = BackgroundMedium.copy(alpha = 0.5f)
    ) {
        Column(modifier = Modifier.padding(Spacing.M)) {
            Text("📊 Текущие настройки:", style = MaterialTheme.typography.titleMedium, color = TextPrimary)
            Spacer(modifier = Modifier.height(Spacing.S))
            
            if (isWheelEnabled) {
                Text("• Колесо удачи: ✅ Включено ($wheelFrequency раз в день)", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
                Text("  Призы: 1️⃣${prizes[0]} • 2️⃣${prizes[1]} • 3️⃣${prizes[2]} • 4️⃣${prizes[3]} • 5️⃣${prizes[4]} • 6️⃣${prizes[5]} 🦄", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
            } else {
                Text("• Колесо удачи: ❌ Отключено", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
            }
            
            Text(if (isTournamentEnabled) "• Семейный турнир: ✅ Включен" else "• Семейный турнир: ❌ Отключен", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
            Text("• Единорог-питомец: 🔒 Всегда включен", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
            Text(if (isUniverseEnabled) "• Единорог-вселенная: ✅ Включена" else "• Единорог-вселенная: ❌ Отключена", style = MaterialTheme.typography.labelSmall, color = TextSecondary)
        }
    }
}

@Composable
private fun SaveButton(onSave: () -> Unit) {
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.ScreenPadding),
        shape = RoundedCornerShape(CornerRadius.Large),
        color = SuccessGreen,
        onClick = onSave
    ) {
        Text("💾 Сохранить настройки", style = MaterialTheme.typography.titleMedium.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold), color = TextPrimary, modifier = Modifier.padding(Spacing.M))
    }
}

File: Screens/GamesParentalControlScreen.swift
import SwiftUI

/// 🎮 Games Parental Control Screen
/// Экран управления геймификацией
/// Источник дизайна: /mobile/wireframes/14c_games_parental_control.html
struct GamesParentalControlScreen: View {
    
    // MARK: - State
    
    @EnvironmentObject private var navigationManager: NavigationManager
    @EnvironmentObject private var localizationManager: LocalizationManager
    @State private var parentAuthMessage: String?
    
    // Сохраняем настройки игр в AppStorage
    @AppStorage("games_wheel_enabled") private var isWheelEnabled: Bool = true
    @AppStorage("games_tournament_enabled") private var isTournamentEnabled: Bool = true
    @AppStorage("games_universe_enabled") private var isUniverseEnabled: Bool = true
    @AppStorage("games_wheel_frequency") private var wheelFrequency: Double = 1
    @AppStorage("games_wheel_prize_1") private var prizeSector1: Double = 5
    @AppStorage("games_wheel_prize_2") private var prizeSector2: Double = 10
    @AppStorage("games_wheel_prize_3") private var prizeSector3: Double = 20
    @AppStorage("games_wheel_prize_4") private var prizeSector4: Double = 50
    @AppStorage("games_wheel_prize_5") private var prizeSector5: Double = 100
    @AppStorage("games_wheel_prize_6") private var prizeSector6: Double = 500
    
    // MARK: - Body
    
    var body: some View {
        ZStack {
            // Фон
            LinearGradient.backgroundGradient
                .ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Навигационная панель
                ALADDINNavigationBar(
                    title: localizationManager.localized("games_parental_nav_title"),
                    subtitle: localizationManager.localized("games_parental_nav_subtitle"),
                    showBackButton: true,
                    showProfileButton: false,
                    showListButton: false,
                    onBack: {
                        navigationManager.goBack()
                    }
                )
                
                // Основной контент
                ScrollView(.vertical, showsIndicators: false) {
                    VStack(spacing: Spacing.l) {
                        // Информация
                        infoCard
                        
                        // Колесо удачи
                        wheelGameCard
                        
                        // Семейный турнир
                        tournamentGameCard
                        
                        // Единорог-питомец (всегда ВКЛ)
                        petGameCard
                        
                        // Единорог-вселенная
                        universeGameCard
                        
                        // Быстрые действия
                        quickActions
                        
                        // Текущие настройки
                        currentSettings
                        
                        // Кнопка сохранить
                        saveButton
                        
                        Spacer()
                            .frame(height: Spacing.xxl)
                    }
                    .padding(.top, Spacing.m)
                }
            }
        }
        .navigationBarHidden(true)
        .alert(
            localizationManager.localized("games_parental_auth_alert_title"),
            isPresented: Binding(
                get: { parentAuthMessage != nil },
                set: { if !$0 { parentAuthMessage = nil } }
            )
        ) {
            Button(localizationManager.localized("common_ok"), role: .cancel) {
                parentAuthMessage = nil
            }
        } message: {
            Text(parentAuthMessage ?? "")
        }
    }
    
    // MARK: - Info Card
    
    private var infoCard: some View {
        HStack(spacing: Spacing.m) {
            Text("💡")
                .font(.system(size: 20))
            
            VStack(alignment: .leading, spacing: Spacing.xs) {
                Text(localizationManager.localized("games_parental_info_title"))
                    .font(.body)
                    .fontWeight(.semibold)
                    .foregroundColor(.textPrimary)
                Text(localizationManager.localized("games_parental_info_description"))
                    .font(.captionSmall)
                    .foregroundColor(.textSecondary)
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
        .padding(Spacing.m)
        .background(
            RoundedRectangle(cornerRadius: CornerRadius.medium)
                .fill(Color.primaryBlue.opacity(0.15))
        )
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    // MARK: - Wheel Game Card
    
    private var wheelGameCard: some View {
        VStack(alignment: .leading, spacing: Spacing.m) {
            // Заголовок с переключателем
            HStack {
                HStack(spacing: Spacing.xs) {
                    Text("🎰")
                        .font(.system(size: 22))
                    Text(localizationManager.localized("games_parental_wheel_title"))
                        .font(.body)
                        .fontWeight(.semibold)
                        .foregroundColor(.textPrimary)
                }
                
                Spacer()
                
                Toggle("", isOn: Binding(
                    get: { isWheelEnabled },
                    set: { newValue in
                        requestParentSessionForSettingChange {
                            isWheelEnabled = newValue
                        }
                    }
                ))
                    .labelsHidden()
            }
            
            // Описание
            Text(localizationManager.localized("games_parental_wheel_description"))
                .font(.captionSmall)
                .foregroundColor(.textSecondary)
                .fixedSize(horizontal: false, vertical: true)
            
            if isWheelEnabled {
                // Частота вращений
                VStack(alignment: .leading, spacing: Spacing.s) {
                    Text(localizationManager.localized("games_parental_wheel_frequency_title"))
                        .font(.caption)
                        .foregroundColor(.textPrimary)
                    
                    Slider(value: $wheelFrequency, in: 1...7, step: 1)
                        .accentColor(.successGreen)
                    
                    HStack {
                        Text(localizationManager.localized("games_parental_wheel_frequency_min"))
                            .font(.captionSmall)
                            .foregroundColor(.textSecondary)
                        Spacer()
                        Text("\(Int(wheelFrequency)) \(wheelFrequency == 1 ? "раз в день" : (wheelFrequency <= 4 ? "раза в день" : "раз в день"))")
                            .font(.caption)
                            .fontWeight(.semibold)
                            .foregroundColor(.successGreen)
                        Spacer()
                        Text(localizationManager.localized("games_parental_wheel_frequency_max"))
                            .font(.captionSmall)
                            .foregroundColor(.textSecondary)
                    }
                }
                
                // Настройка призов
                VStack(alignment: .leading, spacing: Spacing.s) {
                    Text(localizationManager.localized("games_parental_wheel_prizes_title"))
                        .font(.caption)
                        .foregroundColor(.textPrimary)
                    
                    prizeSlider(sector: 1, chance: "40%", prize: $prizeSector1, range: 1...50)
                    prizeSlider(sector: 2, chance: "30%", prize: $prizeSector2, range: 5...100)
                    prizeSlider(sector: 3, chance: "15%", prize: $prizeSector3, range: 10...150)
                    prizeSlider(sector: 4, chance: "10%", prize: $prizeSector4, range: 20...200)
                    prizeSlider(sector: 5, chance: "4%", prize: $prizeSector5, range: 50...300)
                    prizeSlider(sector: 6, chance: "1%", prize: $prizeSector6, range: 100...1000, isJackpot: true)
                    
                    Text(localizationManager.localized("games_parental_wheel_tip"))
                        .font(.captionSmall)
                        .foregroundColor(.textSecondary)
                        .padding(Spacing.s)
                        .background(
                            RoundedRectangle(cornerRadius: CornerRadius.small)
                                .fill(Color.primaryBlue.opacity(0.1))
                        )
                }
            }
        }
        .padding(Spacing.m)
        .background(
            RoundedRectangle(cornerRadius: CornerRadius.large)
                .fill(Color.backgroundMedium.opacity(0.5))
                .opacity(isWheelEnabled ? 1.0 : 0.6)
        )
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    private func prizeSlider(sector: Int, chance: String, prize: Binding<Double>, range: ClosedRange<Double>, isJackpot: Bool = false) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack {
                Text(
                    isJackpot
                    ? localizationManager.localized("games_parental_wheel_sector_jackpot", sector, chance)
                    : localizationManager.localized("games_parental_wheel_sector", sector, chance)
                )
                    .font(.captionSmall)
                    .foregroundColor(.textSecondary)
                Spacer()
                Text("\(Int(prize.wrappedValue)) 🦄")
                    .font(.caption)
                    .fontWeight(.semibold)
                    .foregroundColor(Color(hex: "FFD700"))
            }
            
            Slider(value: prize, in: range, step: 1)
                .accentColor(Color(hex: "FFD700"))
                .frame(height: 4)
        }
    }
    
    // MARK: - Tournament Game Card
    
    private var tournamentGameCard: some View {
        gameCard(
            icon: "🏆",
            title: localizationManager.localized("games_parental_tournament_title"),
            description: localizationManager.localized("games_parental_tournament_description"),
            isEnabled: $isTournamentEnabled
        )
    }
    
    // MARK: - Pet Game Card
    
    private var petGameCard: some View {
        VStack(alignment: .leading, spacing: Spacing.m) {
            HStack {
                HStack(spacing: Spacing.xs) {
                    Text("🦄")
                        .font(.system(size: 22))
                    Text(localizationManager.localized("games_parental_pet_title"))
                        .font(.body)
                        .fontWeight(.semibold)
                        .foregroundColor(.textPrimary)
                }
                
                Spacer()
                
                Text(localizationManager.localized("games_parental_pet_always_on"))
                    .font(.captionSmall)
                    .foregroundColor(Color(hex: "C084FC"))
                    .padding(.horizontal, 10)
                    .padding(.vertical, 5)
                    .background(
                        Capsule()
                            .fill(Color(hex: "A855F7").opacity(0.3))
                            .overlay(
                                Capsule()
                                    .stroke(Color(hex: "C084FC"), lineWidth: 1)
                            )
                    )
            }
            
            Text(localizationManager.localized("games_parental_pet_description"))
                .font(.captionSmall)
                .foregroundColor(.textSecondary)
                .fixedSize(horizontal: false, vertical: true)
        }
        .padding(Spacing.m)
        .background(
            RoundedRectangle(cornerRadius: CornerRadius.large)
                .fill(Color(hex: "A855F7").opacity(0.1))
                .overlay(
                    RoundedRectangle(cornerRadius: CornerRadius.large)
                        .stroke(Color(hex: "C084FC"), lineWidth: 1)
                )
        )
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    // MARK: - Universe Game Card
    
    private var universeGameCard: some View {
        gameCard(
            icon: "🌳",
            title: localizationManager.localized("games_parental_universe_title"),
            description: localizationManager.localized("games_parental_universe_description"),
            isEnabled: $isUniverseEnabled
        )
    }
    
    private func gameCard(icon: String, title: String, description: String, isEnabled: Binding<Bool>) -> some View {
        VStack(alignment: .leading, spacing: Spacing.m) {
            HStack {
                HStack(spacing: Spacing.xs) {
                    Text(icon)
                        .font(.system(size: 22))
                    Text(title)
                        .font(.body)
                        .fontWeight(.semibold)
                        .foregroundColor(.textPrimary)
                }
                
                Spacer()
                
                Toggle("", isOn: Binding(
                    get: { isEnabled.wrappedValue },
                    set: { newValue in
                        requestParentSessionForSettingChange {
                            isEnabled.wrappedValue = newValue
                        }
                    }
                ))
                    .labelsHidden()
            }
            
            Text(description)
                .font(.captionSmall)
                .foregroundColor(.textSecondary)
                .fixedSize(horizontal: false, vertical: true)
        }
        .padding(Spacing.m)
        .background(
            RoundedRectangle(cornerRadius: CornerRadius.large)
                .fill(Color.backgroundMedium.opacity(0.5))
                .opacity(isEnabled.wrappedValue ? 1.0 : 0.6)
        )
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    // MARK: - Quick Actions
    
    private var quickActions: some View {
        VStack(spacing: Spacing.s) {
            Button(action: {
                requestParentSessionForSettingChange {
                    disableAllGames()
                }
            }) {
                Text(localizationManager.localized("games_parental_disable_all"))
                    .font(.body)
                    .fontWeight(.semibold)
                    .foregroundColor(.dangerRed)
                    .frame(maxWidth: .infinity)
                    .padding(Spacing.m)
                    .background(
                        RoundedRectangle(cornerRadius: CornerRadius.medium)
                            .fill(Color.dangerRed.opacity(0.2))
                            .overlay(
                                RoundedRectangle(cornerRadius: CornerRadius.medium)
                                    .stroke(Color.dangerRed, lineWidth: 2)
                            )
                    )
            }
            .buttonStyle(PlainButtonStyle())
            
            Button(action: {
                requestParentSessionForSettingChange {
                    enableAllGames()
                }
            }) {
                Text(localizationManager.localized("games_parental_enable_all"))
                    .font(.body)
                    .fontWeight(.semibold)
                    .foregroundColor(.successGreen)
                    .frame(maxWidth: .infinity)
                    .padding(Spacing.m)
                    .background(
                        RoundedRectangle(cornerRadius: CornerRadius.medium)
                            .fill(Color.successGreen.opacity(0.2))
                            .overlay(
                                RoundedRectangle(cornerRadius: CornerRadius.medium)
                                    .stroke(Color.successGreen, lineWidth: 2)
                            )
                    )
            }
            .buttonStyle(PlainButtonStyle())
        }
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    // MARK: - Current Settings
    
    private var currentSettings: some View {
        VStack(alignment: .leading, spacing: Spacing.s) {
            Text(localizationManager.localized("games_parental_current_settings_title"))
                .font(.body)
                .fontWeight(.semibold)
                .foregroundColor(.textPrimary)
            
            VStack(alignment: .leading, spacing: Spacing.xs) {
                if isWheelEnabled {
                    Text(localizationManager.localized("games_parental_current_wheel_enabled", Int(wheelFrequency)))
                        .font(.captionSmall)
                        .foregroundColor(.textSecondary)
                    Text(localizationManager.localized(
                        "games_parental_current_wheel_prizes",
                        Int(prizeSector1),
                        Int(prizeSector2),
                        Int(prizeSector3),
                        Int(prizeSector4),
                        Int(prizeSector5),
                        Int(prizeSector6)
                    ))
                        .font(.captionSmall)
                        .foregroundColor(.textSecondary)
                } else {
                    Text(localizationManager.localized("games_parental_current_wheel_disabled"))
                        .font(.captionSmall)
                        .foregroundColor(.textSecondary)
                }
                
                Text(
                    isTournamentEnabled
                    ? localizationManager.localized("games_parental_current_tournament_enabled")
                    : localizationManager.localized("games_parental_current_tournament_disabled")
                )
                    .font(.captionSmall)
                    .foregroundColor(.textSecondary)
                
                Text(localizationManager.localized("games_parental_current_pet_always_on"))
                    .font(.captionSmall)
                    .foregroundColor(.textSecondary)
                
                Text(
                    isUniverseEnabled
                    ? localizationManager.localized("games_parental_current_universe_enabled")
                    : localizationManager.localized("games_parental_current_universe_disabled")
                )
                    .font(.captionSmall)
                    .foregroundColor(.textSecondary)
            }
        }
        .padding(Spacing.m)
        .background(
            RoundedRectangle(cornerRadius: CornerRadius.medium)
                .fill(Color.backgroundMedium.opacity(0.5))
        )
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    // MARK: - Save Button
    
    private var saveButton: some View {
        Button(action: {
            requestParentSessionForSettingChange {
                saveSettings()
            }
        }) {
            Text(localizationManager.localized("games_parental_save"))
                .font(.body)
                .fontWeight(.semibold)
                .foregroundColor(.textPrimary)
                .frame(maxWidth: .infinity)
                .padding(Spacing.m)
                .background(
                    LinearGradient(
                        colors: [.successGreen, .successGreen.opacity(0.8)],
                        startPoint: .leading,
                        endPoint: .trailing
                    )
                    .clipShape(RoundedRectangle(cornerRadius: CornerRadius.large))
                )
        }
        .buttonStyle(PlainButtonStyle())
        .padding(.horizontal, Spacing.screenPadding)
    }
    
    // MARK: - Actions
    
    private func disableAllGames() {
        isWheelEnabled = false
        isTournamentEnabled = false
        isUniverseEnabled = false
        
        let generator = UINotificationFeedbackGenerator()
        generator.notificationOccurred(.success)
        print("Все игры отключены (кроме питомца)")
    }
    
    private func enableAllGames() {
        isWheelEnabled = true
        isTournamentEnabled = true
        isUniverseEnabled = true
        
        let generator = UINotificationFeedbackGenerator()
        generator.notificationOccurred(.success)
        print("Все игры включены")
    }
    
    private func saveSettings() {
        let generator = UINotificationFeedbackGenerator()
        generator.notificationOccurred(.success)
        
        print("💾 Настройки сохранены:")
        print("- Колесо: \(isWheelEnabled ? "ВКЛ" : "ВЫКЛ"), частота: \(Int(wheelFrequency))")
        print("- Призы: \(Int(prizeSector1)), \(Int(prizeSector2)), \(Int(prizeSector3)), \(Int(prizeSector4)), \(Int(prizeSector5)), \(Int(prizeSector6))")
        print("- Турнир: \(isTournamentEnabled ? "ВКЛ" : "ВЫКЛ")")
        print("- Вселенная: \(isUniverseEnabled ? "ВКЛ" : "ВЫКЛ")")
        
        navigationManager.goBack()
    }

    @MainActor
    private func requestParentSessionForSettingChange(action: @escaping () -> Void) {
        action()
    }
}

// MARK: - Preview

#if DEBUG
struct GamesParentalControlScreen_Previews: PreviewProvider {
    static var previews: some View {
        GamesParentalControlScreen()
    }
}
#endif



